btrdbextras/opendss_ingest/opendss_ingestor.py

```diff
@@ -0,0 +1,210 @@
+    The simulation uses frequency `fs` and a database connection `conn`.
+
+    Parameters
+    ----------
+    start_ns : int
+        The start time of the simulation in nanoseconds.
+    end_ns : int
+        The end time of the simulation in nanoseconds.
+    collection_prefix : str
+        The prefix for the name of the data collection for the simulation.
+    fs : int, optional
+        The frequency of the simulation, defaults to 30.
+    conn : obj, optional
+        The database connection used for the simulation, defaults to None.
+    model_location : str, optional
+        The file location of the model to be used for the simulation.
+        Defaults to 'Models/13Bus/IEEE13Nodeckt.dss' if not specified.
+
+    """
+    model_location = (model_location
+                      if model_location is not None
+                      else os.path.join(MODEL_REL_PATH,
+                                        'Models/13Bus/IEEE13Nodeckt.dss'))
+    # Initialize simulation parameters
+    load, load_names = initialize_simulation(model_location)
+    timestamps = np.arange(start_ns, end_ns, 1e9 // fs, dtype="int")
+    scale = generate_scaling(1.1, 0.1, [len(load_names), timestamps.size])
+    new_load = scale * load[:, np.newaxis]
+    collections, names, tags, annotations = sims.get_stream_info(
+        base_col=collection_prefix
+    )
+    streams_dict = sims.create_streams(
+        collection_prefix, collections, names, tags, annotations, conn
+    )
+    streamset = StreamSet(list(streams_dict.values()))
+    _stream_info = lambda s: "/".join(
+        [s.collection.replace(
+            collection_prefix + '/', ''
+        ), s.name]
+    )
+    _values = lambda s, cont_event: simulate_event(
+        (V.get(_stream_info(s)) or I.get(_stream_info(s)))[0],
+        cont_event)
+    prev_timestamp = None
+    # For example, event lasts for at most 100 timestamps
+    continue_event_ind_left = np.random.randint(1, 100)
+    # Run simulation
+    for i in range(new_load.shape[1]):
+        V, I = sims.simulate_network(new_load[:, [i]], load_names)
+        if continue_event_ind_left == 0:
+            continue_event_ind_left = np.random.randint(1, 100)
+        else:
+            continue_event_ind_left -= 1
+        now = datetime_to_ns(datetime.utcnow())
+        _datamap_gen = conn._executor.map(
+            lambda s: (s._uuid,
+                       dict(time=now,
+                                   value=_values(s,
+                                                 continue_event_ind_left > 0)
+                       )),
+            streamset._streams
+            )
+        data_map = {}
+        for k,v in _datamap_gen:
+            if v['value'] is not None:
+                data_map[k] = pa.Table.from_pylist([v])
+        _streamset = StreamSet([Stream(conn, uuid=k) for k in
+                                data_map.keys()])
+        _streamset.arrow_insert(data_map)
+        if i % INSERT_BATCH_SIZE == 0:
```

Comment:
What happens if end this loop when this condition is not met? will the data not flush? If so, we should add another condition here.
```suggestion
        if i % INSERT_BATCH_SIZE == 0 or i == new_load.shape[1]-1:
```

Reply:
right now for this, i was just testing for running for a few mins at a time, and wanted points inserted with a specific flush condition. i'm not sure if the `INSERT_BATCH_SIZE` is actually even a good number for flushing.
at the end if not flushed will automatically flushed after 8 hours according to the pqm.